### ISSUE TITLE
@throws missing from autoPagingIterator()

### ISSUE THREAD

**Issue by ZacharyDuBois** [stripe/stripe-php#1678]
### Describe the bug

Right now, there is now `@throws` in the PHPDoc for the `autoPagingIterator()` method. This method makes calls to `nextPage()` and `previousPage()` which subsequently both call `all()` which throws `ApiErrorException`.

This should be added to all of the parent methods to hint when there is an unhandled exception point in IDEs. Right now, when I wrap an `autoPagingIterator()` loop in a `try { } catch() {}` block, there is nothing hinting to the IDE that exceptions can be thrown there.

<img width="669" alt="Screenshot 2024-04-11 at 10 51 48" src="https://github.com/stripe/stripe-php/assets/2458289/0f73fa52-3e25-47d0-a0a0-f6b4645bdcf3">
<img width="891" alt="Screenshot 2024-04-11 at 10 52 02" src="https://github.com/stripe/stripe-php/assets/2458289/5e6f2909-c8a4-4eeb-9b4c-8d23e3e58a3c">

### To Reproduce

Use code snippet below with an IDE with decent PHP language support.

### Expected behavior

IDE to complain when there is an unhandled exception.

<img width="725" alt="Screenshot 2024-04-11 at 10 56 06" src="https://github.com/stripe/stripe-php/assets/2458289/15c78dd7-5084-42ea-a8e4-bb6b74527690">

### Code snippets

```PHP
try {
    foreach ($balanceTransactions->autoPagingIterator() as $balanceTransaction) {
        // Do work
    }
} catch (RateLimitException|ApiErrorException $exception) {
    // IDE thinks this is unreachable/not thrown in anything above.
}
```

### OS

macOS

### PHP version

PHP8.3

### Library version

13.17.0

### API version

2023-10-16

### Additional context

Using PhpStorm 2024.1

**Comment by ZacharyDuBois** [comment]
Related to #605 but different section of the codebase.

**Comment by ramya-stripe** [comment]
Thanks for reporting @ZacharyDuBois 
The fix is out in today's release - version 14.3.0

**Comment by ZacharyDuBois** [comment]
Wooo! Thank you!
